### DISCUSSION TITLE
Branch protection rules also support migration from ADO? didn't work

### DISCUSSION THREAD

**Discussion by maxmanus96** [github/gh-gei#1202] [Q&A] [answered]
Hi everyone. We need to import ADO repos to GH and I used the tool to create migration script and did a demo migration. However, when I checked on ADO there are branch protection rules but those are not migrated to GH, also checked migration logs but no errors, so GH repo shows no branch protection rules. What would be your suggestions in this case? Looking for your help, thanks.

**Comment by timrogers** [comment]
@maxmanus96 Thanks for reaching out! GitHub Enterprise Importer does support migrating some branch protections from ADO - [specifically](https://docs.github.com/en/migrations/using-github-enterprise-importer/migrating-from-azure-devops-to-github-enterprise-cloud/about-migrations-from-azure-devops-to-github-enterprise-cloud#:~:text=Branch%20protections%20for%20the%20repository%20(user%2Dscoped%20branch%20protections%20not%20included)) "branch protections for the repository (user-scoped branch protections not included)". 

Would you be able to share some more details on the branch protections that weren't migrated? If you aren 't comfortable posting them here, feel free to email me at the @github.com email address on my profile.

**Reply by timrogers** [reply]
Following further discussion with @maxmanus96, we believe that this is because they're using cross-repo branch policies, rather than branch protections configured at the repository level. I'm double checking this and will update here once I know more.

**Reply by timrogers** [reply, marked as answer]
👋🏻 GitHub Product Manager here! I've investigated this and I can confirm that the gap is that **we don't support migrating [cross-repo branch policies](https://learn.microsoft.com/en-us/azure/devops/release-notes/2019/repos/sprint-160-update#cross-repo-branch-policy-administration) from ADO**. Only branch policies defined at the repository level are supported. 

We will update our documentation to clarify this, and I've also recorded this gap as a feature request within GitHub.

If other people run into this issue and want these to be migrated, do feel free to comment here so we can learn about customer needs ❤️

**Reply by Strandfelt** [reply]
We would have liked to have this. It's too late now, as we already started the process, though.
